lib/server/ServerDispatcher.js

```diff
@@ -41,6 +38,20 @@ var ServerDispatcher = {
 
 };
 
+function findHandlerForRoute(fragment) {
+    var handler;
+
+    for (var i = 0, len = handlers.length; i < len; i++) {
+        handler = handlers[i];
+
+        if (!handler.route.test(fragment)) {
```

Comment:
Can you invert the logic? It seems that this could have been simpler by just having a `return` rather than a `return` and `continue`